src/framework/core/js/ResourceLoader.js

```diff
@@ -127,33 +127,28 @@ fluid.fetchResources = function (resourceSpecs, callback, options) {
  */
 fluid.fetchResources.explodeForLocales = function (resourceFetcher) {
     fluid.each(resourceFetcher.resourceSpecs, function (resourceSpec) {
-        if (!resourceSpec.launched) {
-            // If options.defaultLocale is set, it will replace any
-            // defaultLocale set on an individual resourceSpec
-            if (resourceFetcher.options.defaultLocale && resourceSpec.defaultLocale === undefined) {
-                resourceSpec.defaultLocale = resourceFetcher.options.defaultLocale;
-            }
-            if (resourceSpec.locale === undefined) {
-                resourceSpec.locale = resourceFetcher.options.locale || resourceSpec.defaultLocale;
-            }
-            resourceSpec.dataType = resourceSpec.dataType || resourceFetcher.options.dataType;
-
-            resourceSpec.loader = fluid.resourceLoader.resolveResourceLoader(resourceSpec);
-            if (!resourceSpec.loader.loader.noPath) {
-                var pathKey = resourceSpec.loader.pathKey;
-                var path = resourceSpec[pathKey];
-                var resolvedPath = resourceSpec[pathKey] = resourceFetcher.transformResourceURL(path);
-                if (resourceSpec.locale) {
-                    resourceSpec.localeExploded = fluid.explodeLocalisedName(resolvedPath, resourceSpec.locale, resourceSpec.defaultLocale);
-                    resourceSpec.localeExplodedSpecs = fluid.transform(resourceSpec.localeExploded, function (oneExploded) {
-                        var togo = {
-                            loader: resourceSpec.loader
-                        };
-                        togo[pathKey] = oneExploded;
-                        return togo;
-                    }, fluid.fetchResources.prepareRequestOptions);
-                }
-            }
+        // If options.defaultLocale is set, it will replace any defaultLocale set on an individual resourceSpec
+        resourceSpec.resolvedDefaultLocale = resourceSpec.defaultLocale || resourceFetcher.options.defaultLocale;
```

Comment:
The comment says, "If options.defaultLocale is set, it will replace any defaultLocale set on an individual resourceSpec". But the code in the next line has resourceSpec.defaultLocale taking precedent over options.defaultLocale. Shall the code be changed?

Reply:
Good catch, it is the comment that is wrong and has been for many years :)